hathor/conf/get_settings.py

```diff
@@ -37,52 +42,71 @@ def HathorSettings() -> Settings:
 
     If neither is set, or if the module import fails, the mainnet configuration is returned.
     """
-    settings_module = get_settings_module()
 
-    if settings_module is not None:
-        log = logger.new()
-        log.warn(
-            "Setting a config module via the 'HATHOR_CONFIG_FILE' env var will be deprecated soon. "
-            "Use the '--config-yaml' CLI option or the 'HATHOR_CONFIG_YAML' env var to set a yaml filepath instead."
-        )
-        settings = getattr(settings_module, 'SETTINGS')
-        assert isinstance(settings, Settings)
-        return settings
+    settings_module_filepath = os.environ.get('HATHOR_CONFIG_FILE')
+    if settings_module_filepath is not None:
+        return _load_settings_singleton(settings_module_filepath, is_yaml=False)
+
+    settings_yaml_filepath = os.environ.get('HATHOR_CONFIG_YAML', conf.MAINNET_SETTINGS_FILEPATH)
+    return _load_settings_singleton(settings_yaml_filepath, is_yaml=True)
+
 
-    settings_filepath = get_settings_filepath()
+def get_settings_module_path() -> Optional[str]:
+    """ Returns the path of the settings module loaded, None if no module was loaded.
 
-    return Settings.from_yaml(filepath=settings_filepath)
+    XXX: Will raise an assertion error if HathorSettings() wasn't used before.
+    """
+    global _settings_singleton
+    assert _settings_singleton is not None, 'HathorSettings() not called before'
+    if _settings_singleton.is_yaml:
+        return None
+    return _settings_singleton.source
 
 
-def get_settings_module() -> Optional[ModuleType]:
-    global _config_file
-    # Import config file for network
-    config_file = os.environ.get('HATHOR_CONFIG_FILE')
-    if _config_file is None:
-        _config_file = config_file
-    elif _config_file != config_file:
-        raise Exception('loading config twice with a different file')
+def get_settings_filepath() -> Optional[str]:
+    """ Returns the path of the YAML settings loaded, None if no YAML file was loaded.
 
-    if not config_file:
+    XXX: Will raise an assertion error if HathorSettings() wasn't used before.
+    """
+    global _settings_singleton
+    assert _settings_singleton is not None, 'HathorSettings() not called before'
+    if not _settings_singleton.is_yaml:
         return None
+    return _settings_singleton.source
```

Comment:
Maybe now we could just replace those two functions with:

```python
def get_settings_source() -> str:
    assert _settings_singleton is not None, 'HathorSettings() not called before'
    return _settings_singleton.source
```

As their only use is in `CliBuilder`:

`settings_source = get_settings_module_path() or get_settings_filepath()` could become just
`settings_source = get_settings_source()`

Reply:
Yeah, makes sense.

Reply:
Done.